<?xml version="1.0"?>
<!--XML-->
<launch>
    <!-- namespace for camera -->
    <arg name="camera_namespace" default="phobos_nuclear"/>

    <arg name="ros_bag_filepath" default="/home/i3dr/i3dr_mapping/data/i3dr_wood_scan/i3dr_ROS_room_scan_2019-07-04-15-45-00[Light-on-Laser-on].bag" />
    <node pkg="rosbag" type="play" name="player" output="screen" args="--clock $(arg ros_bag_filepath)"/>

    <node name="transform_pcl_frame" pkg="i3dr_pcl_tools" type="transform_pcl_frame" output="screen" >
        <param name="tf" type="string" value="world" />
        <param name="pcl2_input" type="string" value="/phobos_nuclear/points2_b" />
        <param name="pcl2_output" type="string" value="/phobos_nuclear/points2_b_map" />
        <param name="pcl_output" type="string" value="/phobos_nuclear/points_b_map" />
    </node>
    
    <node name="pcl_map" pkg="i3dr_pcl_tools" type="pcl_map" output="screen" >
        <param name="resolution" type="double" value="0.01" />
        <param name="pcl2_input" type="string" value="/phobos_nuclear/points2_b_map" />
        <param name="pcl2_output" type="string" value="/map_plc2" />
        <param name="pcl_output" type="string" value="/map_pcl" />
    </node>

    <!-- STEREO MATCHER -->
    <!-- algorithm 0: OpenCV block -->
    <!-- algorithm 1: OpenCV SGBM -->
    <!-- algorithm 2: I3DR SGM -->
    <arg name="stereo_algorithm" default="2" />
    <!-- maxiumum depth in Z -->
    <arg name="depth_max" default="100" />
    <arg name="urdf_namespace" default="$(arg camera_namespace)"/>
    <include file="$(find i3dr_stereo_camera)/launch/stereo_matcher.launch">
        <arg name="use_i3dr_matcher" value="true"/>

        <arg name="run_remote" value="false"/>

        <!-- Matcher Parameters -->
        <!-- see link below for details on matcher parameters -->
        <!-- http://wiki.ros.org/stereo_image_proc/Tutorials/ChoosingGoodStereoParameters -->
        <!-- algorithm 0: OpenCV block -->
        <!-- algorithm 1: OpenCV SGBM -->
        <!-- algorithm 2: I3DR SGM -->
        <arg name="stereo_algorithm" value="$(arg stereo_algorithm)" />

        <!-- maximum depth value (m) -->
        <!-- noise is high in Deimos above 2.5m -->
        <arg name="depth_max" value="$(arg depth_max)" />
        <!--<arg name="depth_min" value="0.1" />-->

        <arg name="camera_namespace" value="$(arg camera_namespace)"/>
        <arg name="frame_id" value="$(arg urdf_namespace)_depth_optical_frame" />
        <arg name="split_laser" value="false" />
    </include>

</launch>